index.js

```diff
@@ -1,291 +1,430 @@
+              switch (req.mainMethod) {
 
-              // We've got a file that's in a subfolder of some length
-              // We only want to list a folder that's directly within the path
-              const subpath = segments[0] + '/'
+                case 'HEAD': {
+                    if(req.mainType){
+                        res.statusCode = 400
+                        res.headers['Content-Length'] = 0
+                    } else {
+                        if(prog.has(req.mainQuery)){
+                            let tempData = prog.get(req.mainQuery)
+                            if(req.mainPath === path.sep){
+                                res.headers['Content-Type'] = req.mainRes
+                                res.headers['Content-Length'] = `${tempData.length}`
+                                res.headers['Accept-Ranges'] = 'bytes'
+                                res.headers['X-Downloaded'] = `${tempData.downloaded}`
+                                res.statusCode = 200
+                            } else {
+                                let foundFile = tempData.files.find(data => {return data.path.endsWith(req.mainPath)})
+                                if(foundFile){
+                                    res.headers['Content-Type'] = getMimeType(req.mainPath)
+                                    res.headers['Content-Length'] = `${foundFile.length}`
+                                    res.headers['Accept-Ranges'] = 'bytes'
+                                    res.headers['X-Downloaded'] = `${foundFile.downloaded}`
+                                    res.statusCode = 200
+                                } else {
+                                    res.statusCode = 400
+                                    res.headers['Content-Length'] = 0
+                                }
+                            }
+                        } else {
+                            res.statusCode = 400
+                            res.headers['Content-Length'] = 0
+                        }
+                    }
+                    break
+                }
+                  
+                case 'GET': {
+                    if(req.mainType){
+                        if(req.mainReq){
+                            res.data = ['<html><head><title>BT-Fetch</title></head><body><div><p>timeout: ' + app._status.timeout + '</p><p>Torrents: ' + app.webtorrent.torrents.length + '</p><p>initial: ' + app._status.initial + '</p><p>current: ' + app._status.current + '</p><p>share: ' + app._status.share + '</p><div></body></html>']
+                        } else {
+                            res.data = [JSON.stringify({timeout: app._timeOut, share: app._status.share, current: app._status.current, initial: app._status.initial, torrents: app.webtorrent.torrents.length})]
+                        }
+                        res.statusCode = 200
+                        res.headers['Content-Type'] = req.mainRes
```

Comment:
fixed, its better not to return this type of information, the user who is using bt-fetch should already know the options beforehand since they are the ones running the module. so i took this part out.